src/app-layout/utils/use-drawers.ts

```diff
@@ -45,41 +45,110 @@ function getToolsDrawerItem(props: ToolsProps): AppLayoutProps.Drawer | null {
   };
 }
 
+const DRAWERS_LIMIT = 2;
+
 function useRuntimeDrawers(
   disableRuntimeDrawers: boolean | undefined,
   activeDrawerId: string | null,
-  onActiveDrawerChange: (newDrawerId: string | null) => void
+  onActiveDrawerChange: (newDrawerId: string | null) => void,
+  activeGlobalDrawersIds: Array<string>,
+  onActiveGlobalDrawersChange: (newDrawerId: string) => void,
+  drawers: AppLayoutProps.Drawer[]
 ) {
-  const [runtimeDrawers, setRuntimeDrawers] = useState<DrawersLayout>({ before: [], after: [] });
-  const onActiveDrawerChangeStable = useStableCallback(onActiveDrawerChange);
+  const [runtimeDrawers, setRuntimeDrawers] = useState<DrawersLayout>({
+    localBefore: [],
+    localAfter: [],
+    global: [],
+  });
+  const onLocalDrawerChangeStable = useStableCallback(onActiveDrawerChange);
+  const onGlobalDrawersChangeStable = useStableCallback(onActiveGlobalDrawersChange);
 
-  const drawerWasOpenRef = useRef(false);
-  drawerWasOpenRef.current = drawerWasOpenRef.current || !!activeDrawerId;
+  const drawersWereOpenRef = useRef(false);
+  drawersWereOpenRef.current = drawersWereOpenRef.current || !!activeDrawerId || !!activeGlobalDrawersIds.length;
 
   useEffect(() => {
     if (disableRuntimeDrawers) {
       return;
     }
     const unsubscribe = awsuiPluginsInternal.appLayout.onDrawersRegistered(drawers => {
-      setRuntimeDrawers(convertRuntimeDrawers(drawers));
-      if (!drawerWasOpenRef.current) {
-        const defaultActiveDrawer = sortByPriority(drawers).find(drawer => drawer.defaultActive);
-        if (defaultActiveDrawer) {
-          onActiveDrawerChangeStable(defaultActiveDrawer.id);
+      const localDrawers = drawers.filter(drawer => drawer.type !== 'global');
+      const globalDrawers = drawers.filter(drawer => drawer.type === 'global');
+      setRuntimeDrawers(convertRuntimeDrawers(localDrawers, globalDrawers));
+      if (!drawersWereOpenRef.current) {
+        const defaultActiveLocalDrawer = sortByPriority(localDrawers).find(drawer => drawer.defaultActive);
+        if (defaultActiveLocalDrawer) {
+          onLocalDrawerChangeStable(defaultActiveLocalDrawer.id);
         }
+
+        const defaultActiveGlobalDrawers = sortByPriority(globalDrawers).filter(drawer => drawer.defaultActive);
+        defaultActiveGlobalDrawers.forEach(drawer => {
+          onGlobalDrawersChangeStable(drawer.id);
+        });
       }
     });
     return () => {
       unsubscribe();
-      setRuntimeDrawers({ before: [], after: [] });
+      setRuntimeDrawers({ localBefore: [], localAfter: [], global: [] });
     };
-  }, [disableRuntimeDrawers, onActiveDrawerChangeStable]);
+  }, [disableRuntimeDrawers, onGlobalDrawersChangeStable, onLocalDrawerChangeStable]);
+
+  useEffect(() => {
+    const unsubscribe = awsuiPluginsInternal.appLayout.onDrawerOpened(drawerId => {
```

Comment:
There are seem to be some issues with this effect. Try adding `console.log` inside `onDrawerOpened` and check if it is rendered too often (for example, on every navigation open/close)

Reply:
Just tried. Works fine

Reply:
I also tried. Does not work fine :(

Reply:
Reproduced offline. Declaring this a **Follow up improvement**